packages/react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,181 @@
+   * Checkbox changes. This handler is called with event and state info.
+   * `(event, { checked, id }) => void`
+   */
+  onChange?: (
+    evt: React.ChangeEvent<HTMLInputElement>,
+    data: { checked: boolean; id: string }
+  ) => void;
+}
+
+const Checkbox = React.forwardRef(
+  (
+    {
+      className,
+      id,
+      labelText,
+      onChange,
+      indeterminate,
+      hideLabel,
+      title = '',
+      ...other
+    }: CheckboxProps,
+    ref
+  ) => {
+    const prefix = usePrefix();
+    const wrapperClasses = classNames(
+      `${prefix}--form-item`,
+      `${prefix}--checkbox-wrapper`,
+      [className]
+    );
+    const innerLabelClasses = classNames(`${prefix}--checkbox-label-text`, {
+      [`${prefix}--visually-hidden`]: hideLabel,
+    });
+
+    return (
+      <div className={wrapperClasses}>
+        <input
+          {...other}
+          type="checkbox"
+          onChange={(evt) => {
+            onChange && onChange(evt, { checked: evt.target.checked, id });
+          }}
+          className={`${prefix}--checkbox`}
+          id={id}
+          ref={(el) => {
+            if (el && indeterminate) {
+              el.indeterminate = indeterminate;
+            }
+            if (typeof ref === 'function') {
+              ref(el);
+            } else if (ref && Object(ref) === ref) {
+              ref.current = el;
+            }
+          }}
+        />
+        <label
+          htmlFor={id}
+          className={`${prefix}--checkbox-label`}
+          title={title}>
+          <Text className={innerLabelClasses}>{labelText}</Text>
+        </label>
+      </div>
+    );
+  }
+);
+
+Checkbox.propTypes = {
+  /**
+   * Specify whether the underlying input should be checked
+   */
+  checked: PropTypes.bool,
+
+  /**
+   * Specify an optional className to be applied to the <label> node
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify whether the underlying input should be checked by default
+   */
+  defaultChecked: PropTypes.bool,
+
+  /**
+   * Specify whether the Checkbox should be disabled
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * Specify whether the label should be hidden, or not
+   */
+  hideLabel: PropTypes.bool,
+
+  /**
+   * Provide an `id` to uniquely identify the Checkbox input
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * Specify whether the Checkbox is in an indeterminate state
+   */
+  indeterminate: PropTypes.bool,
+
+  /**
+   * Provide a label to provide a description of the Checkbox input that you are
+   * exposing to the user
+   */
+  labelText: PropTypes.node.isRequired,
+
+  /**
+   * Provide an optional handler that is called when the internal state of
+   * Checkbox changes. This handler is called with event and state info.
+   * `(event, { checked, id }) => void`
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * Specify a title for the <label> node for the Checkbox
+   */
+  title: PropTypes.string,
```

Comment:
Correct this is intentional. The reason is because since we are doing the spread of `...others` on the input element I have decided to have the interface extend all the native props of an `HTMLInputElement` using `React.InputHTMLAttributes<HTMLInputElement>`. We omit some native props that we are overriding/providing stricter types for